x-pack/metricbeat/tests/integration/setup_integration_test.go

```diff
@@ -0,0 +1,106 @@
+`
+	metricbeat := integration.NewBeat(t, "metricbeat", "../../metricbeat.test")
+	metricbeat.WriteConfigFile(cfg)
+	esURL := integration.GetESURL(t, "http")
+	kURL, _ := integration.GetKibana(t)
+
+	ver, _, _ := strings.Cut(version.GetDefaultVersion(), "-")
+	index := "metricbeat-" + ver
+
+	dataStreamURL, err := integration.FormatDatastreamURL(t, esURL, index)
+	require.NoError(t, err)
+	templateURL, err := integration.FormatIndexTemplateURL(t, esURL, index)
+	require.NoError(t, err)
+	policyURL, err := integration.FormatPolicyURL(t, esURL, index)
+	cleanUpES := func() {
+		_, _, err := integration.HttpDo(t, http.MethodDelete, dataStreamURL)
+		require.NoErrorf(t, err, "cleanup failed: could not remove datastream %s", index)
+		_, _, err = integration.HttpDo(t, http.MethodDelete, templateURL)
+		require.NoErrorf(t, err, "cleanup failed: could not remove index template %s", index)
+		_, _, err = integration.HttpDo(t, http.MethodDelete, policyURL)
+		require.NoErrorf(t, err, "cleanup failed: could not ilm policy %s", index)
+	}
+	// ensure datastream/index template/ ilm policy is set before running the test
+	cleanUpES()
+	t.Cleanup(cleanUpES)
+
+	metricbeat.Start("setup",
+		"--index-management",
+		"-E", "setup.kibana.protocol=http",
+		"-E", "setup.kibana.host="+kURL.Hostname(),
+		"-E", "setup.kibana.port="+kURL.Port(),
+		"-E", "output.elasticsearch.protocol=http",
+		"-E", "output.elasticsearch.hosts=['"+esURL.String()+"']",
+		"-E", "output.file.enabled=false")
+	procState, err := metricbeat.Process.Wait()
+	require.NoError(t, err, "metricbeat setup failed")
+	require.Equal(t, 0, procState.ExitCode(), "metircbeat setup failed: incorrect exit code")
+
+	// generate an event with dynamically mapped fields
+	fields := map[string]string{}
+	totalFields := 500
+	for i := range totalFields {
+		fields[fmt.Sprintf("a-label-%d", i)] = fmt.Sprintf("some-value-%d", i)
+	}
+	event, err := json.Marshal(map[string]any{
+		"@timestamp":        time.Now().Format(time.RFC3339),
+		"kubernetes.labels": fields,
+	})
+	require.NoError(t, err, "could not marshal event to send to ES")
+
+	endpoint := fmt.Sprintf("%s/%s/_doc", esURL.String(), index)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	r, err := http.NewRequestWithContext(ctx, http.MethodPost, endpoint, bytes.NewBuffer(event))
+	require.NoError(t, err, "could not create request to send event to ES")
+	r.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(r)
+	require.NoError(t, err, "could not send request to send event to ES")
+	defer resp.Body.Close()
+
+	failuremsg := fmt.Sprintf("filed to ingest events with %d new fields. If this test fails it likely means the current `index.mapping.total_fields.limit` for metricbeat index (%s) is close to be reached. Check the logs to see why the envent was not ingested", totalFields, index)
+	if !assert.Equal(t, http.StatusCreated, resp.StatusCode, failuremsg) {
```

Comment:
Could use assert.NotEqual here

Reply:
not really. I want to assert it's equal, if the assertion fails, then the extra code runs to help to explain what went wrong